frontend/views/Login.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from "react";
+import { LinearGradient } from "expo-linear-gradient";
+import { Alert, View, StyleSheet } from "react-native";
+import { CommonActions } from "@react-navigation/native";
+import { Button, Card, TextInput } from "react-native-paper";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+
+import { Admin } from "../api/admin";
+import { updateUser } from "../redux/UserAction";
+
+const styles = StyleSheet.create({
+  card: {
+    marginTop: 100,
+    marginLeft: 20,
+    marginRight: 20,
+  },
+  container: {
+    padding: 10,
+    fontSize: 18,
+  },
+});
+
+function Login({ route, navigation, updateUser, user }) {
+  const [name, setName] = React.useState("");
+  const [email, setEmail] = React.useState("");
+  const [password, setPassword] = React.useState("");
+  const [passwordConfirmation, setPasswordConfirmation] = React.useState("");
+
+  const handleLogin = async () => {
+    Admin.login(email, password)
+      .then((response) => {
+        updateUser(response);
+        navigation.dispatch(
+          CommonActions.reset({
+            index: 1,
+            routes: [{ name: "Home" }],
+          })
+        );
+      })
+      .catch((error) => {
+        console.error("Unsuccessful Login", error);
+        Alert.alert(
+          "Unsuccessful Login",
+          "Incorrect username or password",
+          [{ text: "OK", onPress: () => console.log("OK Pressed") }],
+          { cancelable: true }
+        );
+      });
```

Comment:
Personally I prefer the

```javascript
try {
  // do the thing
} catch(err) {
  // handle error
}
```
versus the chaining of `then` and `catch`, but to each their own

Reply:
yeah I'm used to doing it this way, but I'll try the other way